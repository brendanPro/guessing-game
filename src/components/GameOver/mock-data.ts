import type { Pokemon } from "@/types/pokemon";

export const mockPikachu: Pokemon = {
    id: 25,
    name: "pikachu",
    frenchName: "Pikachu",
    sprites: {
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png",
    },
    types: [
      {
        type: {
          name: "electric",
        },
      },
    ],
    height: 4,
    weight: 60,
    stats: [
      {
        base_stat: 35,
        stat: {
          name: "hp",
        },
      },
    ],
  };
  
export const mockCharizard: Pokemon = {
    id: 6,
    name: "charizard",
    frenchName: "Dracaufeu",
    sprites: {
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png",
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png",
    },
    types: [
      {
        type: {
          name: "fire",
        },
      },
      {
        type: {
          name: "flying",
        },
      },
    ],
    height: 17,
    weight: 905,
    stats: [
      {
        base_stat: 78,
        stat: {
          name: "hp",
        },
      },
    ],
  };


export const mockMewtwo: Pokemon = {
  id: 150,
  name: 'mewtwo',
  frenchName: 'Mewtwo',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/150.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/150.png',
  },
  types: [
    {
      type: {
        name: 'psychic',
      },
    },
  ],
  height: 20,
  weight: 1220,
  stats: [
    {
      base_stat: 106,
      stat: {
        name: 'hp',
      },
    },
  ],
};