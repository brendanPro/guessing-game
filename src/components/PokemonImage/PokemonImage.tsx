import React from 'react';
import { type Pokemon } from '@/types/pokemon';
import { Card, CardContent } from '@/components/ui/card';
import { cn } from '@/lib/utils';

interface PokemonImageProps {
  pokemon: Pokemon | null;
  blurLevel: number; // 0-5, where 0 is fully blurred, 5 is clear
  gameOver?: boolean;
  className?: string;
}

export function PokemonImage({ pokemon, blurLevel, gameOver = false, className }: PokemonImageProps) {
  if (!pokemon) {
    return (
      <Card className={cn("w-64 h-64 mx-auto", className)}>
        <CardContent className="flex items-center justify-center h-full">
          <div className="text-muted-foreground">Loading Pokemon...</div>
        </CardContent>
      </Card>
    );
  }

  // Calculate blur intensity (0-20px blur)
  // If game is over, remove all blur
  const blurIntensity = gameOver ? 0 : Math.max(0, 20 - (blurLevel * 4));
  

  return (
    <Card className={cn("w-full max-w-xs sm:max-w-sm mx-auto aspect-square overflow-hidden", className)}>
      <CardContent className="p-4 h-full flex items-center justify-center">
        <div className="relative w-full h-full">
          <img
            src={pokemon.sprites.front_default}
            alt="Pokemon to guess"
            className={cn(
              "w-full h-full object-contain transition-all duration-500",
              "filter"
            )}
            style={{
              filter: `blur(${blurIntensity}px)`,
            }}
          />
          
        </div>
      </CardContent>
    </Card>
  );
}
