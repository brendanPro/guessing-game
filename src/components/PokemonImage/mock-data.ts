import type { Pokemon } from '@/types/pokemon';

export const mockPikachu: Pokemon = {
  id: 25,
  name: 'pikachu',
  frenchName: 'Pikachu',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png',
  },
  types: [
    {
      type: {
        name: 'electric',
      },
    },
  ],
  height: 4,
  weight: 60,
  stats: [
    {
      base_stat: 35,
      stat: {
        name: 'hp',
      },
    },
  ],
};

export const mockCharizard: Pokemon = {
  id: 6,
  name: 'charizard',
  frenchName: 'Dracaufeu',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png',
  },
  types: [
    {
      type: {
        name: 'fire',
      },
    },
    {
      type: {
        name: 'flying',
      },
    },
  ],
  height: 17,
  weight: 905,
  stats: [
    {
      base_stat: 78,
      stat: {
        name: 'hp',
      },
    },
  ],
};

export const mockMewtwo: Pokemon = {
  id: 150,
  name: 'mewtwo',
  frenchName: 'Mewtwo',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/150.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/150.png',
  },
  types: [
    {
      type: {
        name: 'psychic',
      },
    },
  ],
  height: 20,
  weight: 1220,
  stats: [
    {
      base_stat: 106,
      stat: {
        name: 'hp',
      },
    },
  ],
};

export const mockGyarados: Pokemon = {
  id: 130,
  name: 'gyarados',
  frenchName: 'LÃ©viator',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/130.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/130.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/130.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/130.png',
  },
  types: [
    {
      type: {
        name: 'water',
      },
    },
    {
      type: {
        name: 'flying',
      },
    },
  ],
  height: 65,
  weight: 2350,
  stats: [
    {
      base_stat: 95,
      stat: {
        name: 'hp',
      },
    },
  ],
};

