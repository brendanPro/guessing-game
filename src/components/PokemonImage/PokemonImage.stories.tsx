import "@/index.css"
import type { Meta, StoryObj } from '@storybook/react';
import { PokemonImage } from './PokemonImage';
import { mockCharizard, mockGyarados, mockMewtwo, mockPikachu } from "./mock-data";

const meta = {
  title: 'Game/PokemonImage',
  component: PokemonImage,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    pokemon: {
      control: 'object',
      description: 'The Pokemon to display',
    },
    blurLevel: {
      control: { type: 'range', min: 0, max: 5, step: 1 },
      description: 'Blur level from 0 (fully blurred) to 5 (clear)',
    },
    gameOver: {
      control: 'boolean',
      description: 'Whether the game is over (removes all blur)',
    },
  },
} satisfies Meta<typeof PokemonImage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const FullyBlurred: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 0,
    gameOver: false,
  },
};

export const SlightlyBlurred: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 1,
    gameOver: false,
  },
};

export const HalfBlurred: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 2,
    gameOver: false,
  },
};

export const MostlyVisible: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 4,
    gameOver: false,
  },
};

export const FullyVisible: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 5,
    gameOver: false,
  },
};

export const GameOverClear: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 0,
    gameOver: true,
  },
};

export const WithCharizard: Story = {
  args: {
    pokemon: mockCharizard,
    blurLevel: 2,
    gameOver: false,
  },
};

export const WithMewtwo: Story = {
  args: {
    pokemon: mockMewtwo,
    blurLevel: 3,
    gameOver: false,
  },
};

export const WithGyarados: Story = {
  args: {
    pokemon: mockGyarados,
    blurLevel: 1,
    gameOver: false,
  },
};

export const LoadingState: Story = {
  args: {
    pokemon: null,
    blurLevel: 0,
    gameOver: false,
  },
};

export const BlurProgression: Story = {
  args: {
    pokemon: mockPikachu,
    blurLevel: 0,
    gameOver: false,
  },
  render: (args) => (
    <div className="flex flex-col gap-4">
      <div className="text-center">
        <p className="text-sm font-semibold mb-2">Blur Level 0 (Fully Blurred)</p>
        <PokemonImage {...args} blurLevel={0} />
      </div>
      <div className="text-center">
        <p className="text-sm font-semibold mb-2">Blur Level 2</p>
        <PokemonImage {...args} blurLevel={2} />
      </div>
      <div className="text-center">
        <p className="text-sm font-semibold mb-2">Blur Level 5 (Clear)</p>
        <PokemonImage {...args} blurLevel={5} />
      </div>
    </div>
  ),
};

