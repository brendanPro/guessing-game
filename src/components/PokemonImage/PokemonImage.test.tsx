import { describe, test, expect } from "bun:test";
import { renderToStaticMarkup } from "react-dom/server";
import { PokemonImage } from "./PokemonImage";
import { mockCharizard, mockPikachu, mockMewtwo, mockGyarados } from './mock-data';


describe("PokemonImage Component", () => {
  test("snapshot with Pikachu fully blurred", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={0} gameOver={false} />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Pikachu half blurred", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={2} gameOver={false} />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Pikachu fully visible", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={5} gameOver={false} />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Pikachu game over", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={0} gameOver={true} />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Charizard", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockCharizard} blurLevel={2} gameOver={false} />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with null pokemon (loading state)", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={null} blurLevel={0} gameOver={false} />
    );
    expect(html).toMatchSnapshot();
  });

  test("displays loading state when pokemon is null", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={null} blurLevel={0} gameOver={false} />
    );
    expect(html).toContain("Loading Pokemon...");
  });

  test("displays pokemon image when pokemon is provided", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={0} gameOver={false} />
    );
    expect(html).toContain(mockPikachu.sprites.front_default);
  });

  test("has correct alt text", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={0} gameOver={false} />
    );
    expect(html).toContain('alt="Pokemon to guess"');
  });

  test("applies blur filter when blurLevel is 0", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={0} gameOver={false} />
    );
    expect(html).toContain("blur(20px)");
  });

  test("applies less blur when blurLevel is 2", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={2} gameOver={false} />
    );
    expect(html).toContain("blur(12px)");
  });

  test("applies minimal blur when blurLevel is 4", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={4} gameOver={false} />
    );
    expect(html).toContain("blur(4px)");
  });

  test("applies no blur when blurLevel is 5", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={5} gameOver={false} />
    );
    expect(html).toContain("blur(0px)");
  });

  test("removes all blur when gameOver is true", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={0} gameOver={true} />
    );
    expect(html).toContain("blur(0px)");
  });

  test("displays Charizard sprite correctly", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockCharizard} blurLevel={2} gameOver={false} />
    );
    expect(html).toContain(mockCharizard.sprites.front_default);
  });

  test("displays Mewtwo sprite correctly", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockMewtwo} blurLevel={3} gameOver={false} />
    );
    expect(html).toContain(mockMewtwo.sprites.front_default);
  });

  test("displays Gyarados sprite correctly", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockGyarados} blurLevel={1} gameOver={false} />
    );
    expect(html).toContain(mockGyarados.sprites.front_default);
  });

  test("has transition classes for smooth blur animation", () => {
    const html = renderToStaticMarkup(
      <PokemonImage pokemon={mockPikachu} blurLevel={2} gameOver={false} />
    );
    expect(html).toContain("transition-all");
    expect(html).toContain("duration-500");
  });
});

