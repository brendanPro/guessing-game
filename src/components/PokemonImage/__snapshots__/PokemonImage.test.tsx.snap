// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImage Component snapshot with Pikachu fully blurred 1`] = `"<link rel="preload" as="image" href="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"/><div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm w-full max-w-xs sm:max-w-sm mx-auto aspect-square overflow-hidden"><div data-slot="card-content" class="p-4 h-full flex items-center justify-center"><div class="relative w-full h-full"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png" alt="Pokemon to guess" class="w-full h-full object-contain transition-all duration-500 filter" style="filter:blur(20px)"/></div></div></div>"`;

exports[`PokemonImage Component snapshot with Pikachu half blurred 1`] = `"<link rel="preload" as="image" href="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"/><div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm w-full max-w-xs sm:max-w-sm mx-auto aspect-square overflow-hidden"><div data-slot="card-content" class="p-4 h-full flex items-center justify-center"><div class="relative w-full h-full"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png" alt="Pokemon to guess" class="w-full h-full object-contain transition-all duration-500 filter" style="filter:blur(12px)"/></div></div></div>"`;

exports[`PokemonImage Component snapshot with Pikachu fully visible 1`] = `"<link rel="preload" as="image" href="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"/><div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm w-full max-w-xs sm:max-w-sm mx-auto aspect-square overflow-hidden"><div data-slot="card-content" class="p-4 h-full flex items-center justify-center"><div class="relative w-full h-full"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png" alt="Pokemon to guess" class="w-full h-full object-contain transition-all duration-500 filter" style="filter:blur(0px)"/></div></div></div>"`;

exports[`PokemonImage Component snapshot with Pikachu game over 1`] = `"<link rel="preload" as="image" href="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"/><div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm w-full max-w-xs sm:max-w-sm mx-auto aspect-square overflow-hidden"><div data-slot="card-content" class="p-4 h-full flex items-center justify-center"><div class="relative w-full h-full"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png" alt="Pokemon to guess" class="w-full h-full object-contain transition-all duration-500 filter" style="filter:blur(0px)"/></div></div></div>"`;

exports[`PokemonImage Component snapshot with Charizard 1`] = `"<link rel="preload" as="image" href="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png"/><div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm w-full max-w-xs sm:max-w-sm mx-auto aspect-square overflow-hidden"><div data-slot="card-content" class="p-4 h-full flex items-center justify-center"><div class="relative w-full h-full"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png" alt="Pokemon to guess" class="w-full h-full object-contain transition-all duration-500 filter" style="filter:blur(12px)"/></div></div></div>"`;

exports[`PokemonImage Component snapshot with null pokemon (loading state) 1`] = `"<div data-slot="card" class="bg-card text-card-foreground rounded-xl border shadow-sm w-64 h-64 mx-auto"><div data-slot="card-content" class="p-6 pt-0 flex items-center justify-center h-full"><div class="text-muted-foreground">Loading Pokemon...</div></div></div>"`;
